private/pkg/git/git.go

```diff
@@ -349,3 +254,59 @@ type TreeNode interface {
 	// Mode is the file mode of the object referenced by this Node.
 	Mode() ObjectMode
 }
+
+// Repository is a git repository that is backed by a `.git` directory.
+type Repository interface {
+	// BaseBranch is the base branch of the repository. This is either configured
+	// via the `OpenRepositoryWithBaseBranch` option, or discovered via the remote
+	// named `origin`. Therefore, discovery requires that the repository is pushed
+	// to the remote.
```

Comment:
Ideally we auto discover as much as we can, but in the cases where we can't, I can see a `buf sync --main-branch develop` to help with this.

Reply:
Should this be a flag or a config in `buf.yaml`? It feels to me like this is a reasonable use-case for extending the config file since we already have BSR repository related config (e.g. `name`) in the config.

Reply:
I think when this came up at some point, we talked about storing refspecs in the `buf.yaml` and it was shot down? It was before my time, but I remember reading it and now I can't find it.

I think this is not module config, this is sync config. We already somewhat ignore the `name` in `buf.yaml` in sync world (because names can change over time), which I think is step in the right direction.

Generally, the less config we have, the more we've won with `buf sync`.

Reply:
Fair enough -- I am onboard with this approach :)